### PR TITLE
Implementing X icon in search components

### PR DESCRIPTION
**Pull request by e34lin**
Added an X icon for the search components.

Reference: https://docs.google.com/document/d/1GLshBv7Ue09ceCFDSLmFvrgbEofGkY6R0jXlKxVjqHc/edit number 13 under organization

### CODE REVIEW

**Comment by coverbeck** [comment]
Screenshots would be helpful to get a sense of what it looks like.

**Review by coverbeck — COMMENTED** [review]
I think the X should be vertically centered. Or maybe bottom-aligned. Right now it's top-aligned and looks off to me (Firefox, Mac OSx -- see screenshot), but I'll defer to others.

It's hard to tell in the doc how it should be.


Screen Shot 2021-05-25 at 4 22 50 PM

**Comment by garyluu** [comment]
> I think the X should be vertically centered. Or maybe bottom-aligned. Right now it's top-aligned and looks off to me (Firefox, Mac OSx -- see screenshot), but I'll defer to others.
> 
> It's hard to tell in the doc how it should be.
> 
> Screen Shot 2021-05-25 at 4 22 50 PM

Should probably look like the first example: https://material.angular.io/components/input/examples

File: src/app/home-page/home-logged-out/home.component.html
Comment (garyluu) [inline src/app/home-page/home-logged-out/home.component.html:57]: (keyup.enter) uses search.value. this addition uses value. pick one and use it for both features

File: src/app/search/basic-search/basic-search.component.scss
Comment (garyluu) [inline src/app/search/basic-search/basic-search.component.scss:12]: need explanation on why this search bar has different styling than everything else on the site

**Comment by e34lin** [comment]
> Screenshots would be helpful to get a sense of what it looks like.

Screen Shot 2021-05-25 at 9 16 35 PM
Screen Shot 2021-05-25 at 9 16 44 PM
Screen Shot 2021-05-25 at 9 20 02 PM

**Comment by e34lin** [comment]
> I think the X should be vertically centered. Or maybe bottom-aligned. Right now it's top-aligned and looks off to me (Firefox, Mac OSx -- see screenshot), but I'll defer to others.
> 
> It's hard to tell in the doc how it should be.
> 
> Screen Shot 2021-05-25 at 4 22 50 PM

It looks vertically aligned on my localhost, but let me take another look at it, thank you!

File: src/app/home-page/home-logged-out/home.component.html
Comment (e34lin) [reply src/app/home-page/home-logged-out/home.component.html:57]: okay, didn't catch that (keyup.enter) uses search.value, will make the change thank you!

File: src/app/search/basic-search/basic-search.component.scss
Comment (e34lin) [reply src/app/search/basic-search/basic-search.component.scss:12]: The search bar previously had a class of "w-100 mb-3" so I added the margin bottom in the styling and the alignment for the button.

Comment (garyluu) [reply src/app/search/basic-search/basic-search.component.scss:12]: so i think the implementation is going backwards. .form-field should be removed, this .search-position should be removed. the w-100 and mb-3 should be removed. And a w-100 should be added to the form field.  and mat-label should be added.

This should result in a smaller form field which is more inline with the mockups and this also has the advantage of having little to no css.

**Comment by e34lin** [comment]
Screen Shot 2021-05-28 at 3 28 27 PM
Screen Shot 2021-05-28 at 3 30 15 PM

File: src/app/home-page/home-logged-out/home.component.html
Comment (garyluu) [inline src/app/home-page/home-logged-out/home.component.html:57]: not entirely sure this is needed.

File: src/app/home-page/home-logged-out/home.component.scss
Comment (garyluu) [inline src/app/home-page/home-logged-out/home.component.scss:114]: try to put the button without the close-alignment classes in any component. https://stackblitz.com/angular/nlkqgebarxk?file=src%2Fapp%2Finput-clearable-example.ts has no styling and looks close to what it should be.

File: src/app/search/basic-search/basic-search.component.html
Comment (garyluu) [inline src/app/search/basic-search/basic-search.component.html:13]: i prefer the other method without ngModel. ngModel is for two-way binding which isn't needed here

File: src/styles.scss
Comment (garyluu) [inline src/styles.scss:1490]: it's likely the absolute position is making things worse

File: src/styles.scss
Comment (garyluu) [inline src/styles.scss:1485]: if you're finding that the x is disappearing without the padding-right, it's likely that the width of the search-form-field is wrong

**Review by garyluu — COMMENTED** [review]
In general, try to use less styling.

File: src/app/home-page/home-logged-out/home.component.scss
Comment (e34lin) [reply src/app/home-page/home-logged-out/home.component.scss:114]: This is how it looks like when I don't do any additional styling for the button
Screen Shot 2021-05-28 at 4 52 15 PM

File: src/styles.scss
Comment (e34lin) [reply src/styles.scss:1485]: I added the padding so when a long input is passed it doesn't cover the x icon

Without padding:
Screen Shot 2021-05-28 at 4 55 37 PM

File: src/styles.scss
Comment (e34lin) [reply src/styles.scss:1490]: I added the absolute position because I found on firefox and chrome the x icons were showing up in different places

File: src/app/home-page/home-logged-out/home.component.html
Comment (e34lin) [reply src/app/home-page/home-logged-out/home.component.html:57]: If I delete this line the x icon only shows up the first time I enter something in the input

File: src/app/search/basic-search/basic-search.component.html
Comment (e34lin) [reply src/app/search/basic-search/basic-search.component.html:13]: Sorry, which other method are you referring to?

File: src/styles.scss
Comment (garyluu) [reply src/styles.scss:1485]: With 0 styling: 
image

File: src/app/home-page/home-logged-out/home.component.scss
Comment (garyluu) [reply src/app/home-page/home-logged-out/home.component.scss:114]: figure out what styling is messing with it and remove that style instead of adding more styles

File: src/app/home-page/home-logged-out/home.component.html
Comment (garyluu) [reply src/app/home-page/home-logged-out/home.component.html:57]: odd...but anyways, assuming you always want the x to show up (i prefer not), try removing this and the *ngIf on the button then

Comment (e34lin) [reply src/app/home-page/home-logged-out/home.component.html:57]: okay, thank you I'll try that out. The x doesn't always show, just only when you enter something into the search

File: src/app/home-page/home-logged-out/home.component.scss
Comment (e34lin) [reply src/app/home-page/home-logged-out/home.component.scss:114]: okay, thank you. will try to figure it out

File: src/app/home-page/home-logged-out/home.component.scss
Comment (garyluu) [inline src/app/home-page/home-logged-out/home.component.scss:122]: if you can't figure out how to not have this still, put a TODO along with your best guess on what the problem is

**Review by garyluu — APPROVED** [review]

**Review by denis-yuen — CHANGES_REQUESTED** [review]
Heads-up, https://github.com/dockstore/dockstore-ui2/pull/1243/commits/50ee7159f14872d94f6449df98c59af842a6bf2a looks like it is taking in the angular 11 changes from develop.

Separately, let's try to get this in by end of day

Edit to add: chatting with @garyluu 
looks like the workaround without upgrading angular is `display: inline-block` with a good comment to remove
Follow-up with a PR to develop after this is merged to the release branch

**Review by coverbeck — APPROVED** [review]

File: src/styles.scss
Comment (garyluu) [inline src/styles.scss:1523]: umm....if it's a style that's specific to a component (because it can't be generalized), then it should be in the component scss. In this case, there's some weird problems going on and it's unlikely to be occur in a future component so we won't ever use the class elsewhere.

**Review by denis-yuen — APPROVED** [review]
Angular 11 looks gone, thanks!

**Pull request merged**
